src/app/qgisapp.cpp

```diff
@@ -12344,6 +12335,19 @@ QMenu *QgisApp::createPopupMenu()
   return menu;
 }
 
+void QgisApp::notifyUser( QString title, QString message )
```

Comment:
Thanks for reviewing, I'll make the changes.
For the native system wrapper, I'm not really into that kind of development, but according to what briefly discussed with @dakcarto and @NathanW2 that will be probably be the next move. 
For now, the limited scope of this PR is exposing for a (limited!) broader use the tray notifications that are already implemented in master but only available for the task completion.